### ISSUE TITLE
 Incorrect signature warning (coroutines)

### ISSUE THREAD

**Issue by tomas-jezek** [microsoft/Microsoft.Unity.Analyzers#113]
#### Bug description
- Version: 1.8.2
- Analyzer rule: [UNT0006](https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/f952c76dc147a9163220c61953e8ff2550f040c4/doc/UNT0006.md)
- Warning: This Unity message uses an incorrect method signature.

The warning works well enough for most Unity messages. 
But some of them can be used as coroutines and the aforementioned warning results in some false positives with the less commonly used messages (for example OnPostRender, OnPreRender). 
More common messages (Start, OnMouseDown, OnTriggerStay) work fine.

#### To reproduce
Steps or code to reproduce the behaviour:
1.　Declare the following method, which the analyzers accept
```csharp
void OnPostRender() {　}
```
2.　Change the signature
```csharp
IEnumerator OnPostRender() {　yield break;　}
```
#### Expected behaviour
To behave similarly to more common Unity messages, and accept all possible signatures.
```csharp
void Start() {　}
IEnumerator Start() {　yield break;　}
```

#### Screenshots
![OnPostRender - coroutine](https://user-images.githubusercontent.com/18735072/96800871-ad9c6800-1406-11eb-9a9b-c6668d845ffb.png)

#### Additional suggestion
Some Unity messages can be coroutines, some cannot.
Perhaps it would be helpful to add an additional, more specific warning, i.e. _"The message '**Update**' cannot be a coroutine"_ instead of UNT0006 in some cases, to make a distinction between a complete mistake:
```csharp
void Update(int foo) {　}
```
and a merely misguided attempt to use it in a way which won't work.
```csharp
IEnumerator Update() {　yield break;　}
```

**Comment by sailro** [comment]
See https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/main/src/Microsoft.Unity.Analyzers/UnityStubs.cs#L167

```csharp
IEnumeratorOrVoid Start() { return null; }
```

but

```csharp
void OnPostRender() { }
```

I believe this is just the way we declare Unity `stubs`. We have to audit them to fix return type accordingly.

**Comment by sailro** [comment]
I fixed the main issue in the related PR.

Regarding the additional suggestion, yes this can be useful, let me think about it.
